crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -341,6 +341,13 @@ impl<'db> ClassType<'db> {
             }
         }
 
+        if self
+            .iter_mro(db)
+            .any(|base| matches!(base, ClassBase::Dynamic(DynamicType::Any)))
+        {
+            return true;
```

Comment:
Is this generally true, or do we need some exceptions for `@final` classes here?
```py
from typing import Any, final

class SomeClass: ...

@final
class FinalClass: ...

class Mock(Any): ...

x: SomeClass = Mock()
y: FinalClass = Mock()  # okay or not?
```

Reply:
I think some exceptions for `@final` would make sense here!

Reply:
Done